src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,8 +50,17 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+//            if (!Collections.disjoint(wordsInName, keywords)) {
+//                matchedPersons.add(person);
+//            }
+            for (String nameWord : wordsInName) {
+                for (String keyword : keywords) {
+                    if (nameWord.equalsIgnoreCase(keyword)) {
```

Comment:
This increases the complexity of your function as you have more `for` loops. You may want add a new method that converts `wordsInName` and `keywords` separately, while retaining the `disjoint` function that already checks for the matching between keyword and name for you.  Read up about `Collections.disjoint` here:  https://www.geeksforgeeks.org/java-util-collections-disjoint-method-java-examples/.